addons/website_quote/static/src/js/website_quotation.js

```diff
@@ -17,26 +17,97 @@ if(!$('.o_website_quote').length) {
         },
         onClick: function(ev){
             ev.preventDefault();
-            var self = this;
             var href = this.$el.attr("href");
-            var order_id = href.match(/order_id=([0-9]+)/);
-            var line_id = href.match(/update_line\/([0-9]+)/);
-            var token = href.match(/token=(.*)/);
-            ajax.jsonRpc("/quote/update_line", 'call', {
-                'line_id': line_id[1],
-                'order_id': parseInt(order_id[1]),
-                'token': token[1],
-                'remove': self.$el.is('[href*="remove"]'),
-                'unlink': self.$el.is('[href*="unlink"]')
-            }).then(function (data) {
-                if(!data){
-                    location.reload();
-                }
-                self.$el.parents('.input-group:first').find('.js_quantity').val(data[0]);
-                $('[data-id="total_amount"]>span').html(data[1]);
-            });
+            var order_id = href.match(/order_id=([0-9]+)/)[1];
+            var line_id = href.match(/update_line(_dict)?\/([0-9]+)/)[2];
+            var token = href.match(/token=([\w\d-]*)/)[1];
+
+            var callParams = {
+                'line_id': line_id,
+                'order_id': parseInt(order_id),
+                'token': token,
+                'remove': this.$el.is('[href*="remove"]'),
+                'unlink': this.$el.is('[href*="unlink"]')
+            };
+            this._callUpdateLineRoute(callParams).then(this._updateOrderValues.bind(this));
             return false;
         },
+        /**
+         * Only for compatibily reasons
+         * when server is not restarted
+         * and the route update_line_dict doesn't exist
+         */
+        _callUpdateLineRoute: function (params) {
+            var def = new $.Deferred();
+            ajax.jsonRpc("/quote/update_line_dict", 'call', params)
+                .then(def.resolve.bind(def))
+                .fail(function () {
+                    // Compatibility: the server may not have been restarted
+                    // So the real route may not exist
+                    ajax.jsonRpc("/quote/update_line", 'call', params)
+                        .then(def.resolve.bind(def))
+                        .fail(def.reject.bind(def));
+                });
+            return def;
+        },
+        _updateOrderValues: function (data) {
```

Comment:
I would make sure this method receives proper data to keep it clean, and handle the "bad case" fallback in the compatibility method above. This will also allow to remove the route compatibility code more easily in master if it's in only one place.

Also ideally there should be a proper docstring explaining what it does and the what are expected values in `data`.